### DISCUSSION TITLE
Why do I still encounter CORS issues when requesting functions despite adding a Web Platform and setting the host name?

### DISCUSSION THREAD

**Discussion by OneCoser** [appwrite/appwrite#8079] [Q&A] [answered]
### 👟 Reproduction steps

<img width="656" alt="image" src="https://github.com/appwrite/appwrite/assets/6349707/0d78153c-8c34-4cd4-a51c-7a2533b6f8e2">

<img width="502" alt="image" src="https://github.com/appwrite/appwrite/assets/6349707/b0f7ef47-110f-495e-ac8c-2eee7e243cb3">

### 👍 Expected behavior

Request successful

### 👎 Actual Behavior

Request error

### 🎲 Appwrite version

Appwrite Cloud

### 💻 Operating system

MacOS

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment, marked as answer]
@OneCoser, thanks for raising this issue! 🙏🏼 When you execute a function via function domain, the request goes straight to your function and the headers you return are specified by your function. If your function is not returning the relevant CORS headers, you'll see this error. 

If you want to make use of Appwrite's permissions and CORS handling, it would be best to call the [Create execution API](https://appwrite.io/docs/references/cloud/client-web/functions#createExecution).
